src/main/java/com/company/sample/application/CreateOrderThread.java

```diff
@@ -61,6 +61,10 @@ public void createOrder(String productName){
 			}
 
 			SalesSystem.orders.put(orderDate, order);
+                        //Check if the Order entered and present
+                        if (SalesSystem.orders.containsKey(orderDate)) {
+                                System.out.println("New order verified to be present in hashmap: " + SalesSystem.orders.get(orderDate)); 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer



**Problem**  
 You are using a `ConcurrentHashMap`, but your usage of `containsKey()` and `get()` may not be thread-safe at lines: **65 and 66**. In between the check and the `get()` another thread can remove the key and the `get()` will return `null`. The remove that can remove the key is at line: **59**.

**Fix**  
 Consider calling `get()`, checking instead of your current check if the returned object is `null`, and then using that object only, without calling `get()` again.

**More info**  
 [View an example on GitHub](https://github.com/apache/hadoop/blob/f16cf877e565084c66bc63605659b157c4394dc8/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/S3Guard.java#L302-L304) (external link).